app/helpers/dashboard_helper.rb

```diff
@@ -15,6 +15,8 @@ def dashboard_actions
 
   def render_widget(widget)
     render(:partial => widget.template, :locals => widget.data)
+  rescue ActionView::MissingTemplate => e
+    _("Missing template for widget '%s'. The widget or a plugin providing it was probably removed") % widget.name
```

Comment:
I think "Widget cannot be found" is a bit more precise and less hesitant ("probably removed"). Can you raise a Foreman::Exception so that there's an ERF code for that, so that when this is seen, people can search for it and find possible causes for it in the wiki?